dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/ValueMapFlattener.java

```diff
@@ -0,0 +1,114 @@
+package no.unit.nva.dynamodb;
+
+import com.amazonaws.services.lambda.runtime.events.models.dynamodb.AttributeValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+
+public class ValueMapFlattener {
```

Comment:
This is an example of why this PR is not minimal. One could have implemented the indexing service for only flat documents for example and then introduce the flattening. The code in this class is very complex and needs a lot of attention to details. No testing exists for the functionality of this class.

Reply:
The ValueMapFlattener has it's own task.. going to be changed in NP-1301. Will be changed to a xxTransformer